src/lib.rs

```diff
@@ -10,15 +10,34 @@
 
 extern crate cc3200_sys;
 
+#[macro_use]
 pub mod cc3200;
 pub mod isr_vectors;
 
-// We need to define the panic_fmt "lang item", which is just a function. This specifies
-// what the program should do when a `panic!` occurs. Our program won't panic, so we can leave the
-// function body empty for now.
+// These functions are used by the compiler, but not  are normally provided by libstd.
+#[allow(private_no_mangle_fns)]
 mod lang_items {
+    use core::fmt::Arguments;
+
+    #[lang = "eh_personality"]
+    #[no_mangle]
+    pub extern fn rust_eh_personality() {
+    }
+
+    // This function may be needed based on the compilation target.
+    #[lang = "eh_unwind_resume"]
+    #[no_mangle]
+    pub extern fn rust_eh_unwind_resume() {
+    }
+
     #[lang = "panic_fmt"]
-    extern "C" fn panic_fmt() {}
+    #[no_mangle]
+    pub extern fn rust_begin_panic(_msg: Arguments,
+                                   _file: &'static str,
+                                   _line: u32) -> ! {
+        println!("Panic at {}:{} : {}", _file, _line, _msg);
+        loop {}
```

Comment:
The infinite loop will only stop the current thread. Is that what we want? Or should it stop all threads?

Perhaps we need a weak function to call so that the app can make a descision?